test_scripts/API/VehicleData/OnVehicleData/008_OnVD_enum_and_bool_values.lua

```diff
@@ -0,0 +1,37 @@
+----------------------------------------------------------------------------------------------------
+-- Description: Check that SDL processes OnVehicleData notification with <vd_param> parameter
+--
+-- Preconditions:
+-- 1) SDL and HMI are started
+-- 2) SubscribeVehicleData, OnVehicleData RPCs and <vd_param> parameter are allowed by policies
+-- 3) App is registered
+-- 4) App is subscribed to <vd_param> parameter data
+--
+-- In case:
+-- 1) HMI sends valid OnVehicleData notification with <vd_param> parameter data to SDL
+-- (iterate trough all possible enum and boolean values of applicable parameters)
+-- SDL does:
+-- - a) transfer this notification to App
+-- Exception: Notification for unsubscribable VD parameter is not transfered
+----------------------------------------------------------------------------------------------------
+--[[ Required Shared libraries ]]
+local common = require('test_scripts/API/VehicleData/common')
+
+--[[ Local Constants ]]
+local testTypes = {
+  common.testType.ENUM_ITEMS,
+  common.testType.BOOL_ITEMS
+}
+
+--[[ Scenario ]]
+common.Title("Preconditions")
+common.Step("Clean environment and update preloaded_pt file", common.preconditions)
+common.Step("Start SDL, HMI, connect Mobile, start Session", common.start)
+common.Step("Register App", common.registerApp)
+common.Step("Activate App", common.activateApp)
+
+common.Title("Test")
+common.getTestsForOnVD(testTypes)
+
+common.Title("Postconditions")
```

Comment:
Parameter names that are not checked can be removed.

Reply:
This is done intentionally to be sure that some test is not applicable for particular VD parameter.